backend/internal/system/crypto/algorithm.go

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package crypto
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+)
+
+// CryptoAlgorithm defines the interface that all encryption algorithms must implement
+type CryptoAlgorithm interface {
+	Encrypt(key []byte, plaintext []byte) (string, error)
+	Decrypt(key []byte, encodedData string) ([]byte, error)
+	Name() Algorithm
+}
+
+// AESGCMAlgorithm implements the CryptoAlgorithm interface for AES-256-GCM
+type AESGCMAlgorithm struct{}
+
+// Encrypt encrypts plaintext using AES-256-GCM and returns a serialized EncryptedData
+func (a *AESGCMAlgorithm) Encrypt(key []byte, plaintext []byte) (string, error) {
+	// Create AES cipher
+	block, err := aes.NewCipher(key)
+	if err != nil {
+		return "", err
+	}
+
+	// Create GCM mode
+	aesgcm, err := cipher.NewGCM(block)
+	if err != nil {
+		return "", err
+	}
+
+	// Create a nonce
+	nonce := make([]byte, aesgcm.NonceSize())
+	if _, err = rand.Read(nonce); err != nil {
+		return "", err
+	}
+
+	// Encrypt and authenticate plaintext, prepend nonce
+	ciphertext := aesgcm.Seal(nonce, nonce, plaintext, nil)
+
+	// Create metadata structure
+	encData := EncryptedData{
+		Algorithm:  AESGCM,
+		Ciphertext: base64.StdEncoding.EncodeToString(ciphertext),
+	}
+
+	// Serialize to JSON
+	jsonData, err := json.Marshal(encData)
+	if err != nil {
+		return "", err
+	}
+
+	return base64.StdEncoding.EncodeToString(jsonData), nil
```

Comment:
```suggestion
	return string(jsonData), nil
```
Do we need to encode the json object again? From the crypto perspective I don't see a reason

Reply:
+1.. Keep the json without encoding

Reply:
Fixed in https://github.com/asgardeo/thunder/pull/85/commits/b63792f6d32823589f164f3fd74a2b0e44ac021f